src/js/DropContainer.jsx

```diff
@@ -0,0 +1,281 @@
+
+   onDragStart(e) {
+      if (this.props.onDragStart) {
+         this.props.onDragStart.call(this, e);
+      }
+   }
+
+   onDragEnter(e) {
+      e.preventDefault();
+
+      // Count the dropzone and any children that are entered.
+      ++this.enterCounter;
+
+      // This is tricky. During the drag even the dataTransfer.files is null
+      // But Chrome implements some drag store, which is accesible via dataTransfer.items
+      const dataTransferItems = this.extractData(e);
+
+      // Now we need to convert the DataTransferList to Array
+      const allFilesAccepted = this.fileAccepted(Array.prototype.slice.call(dataTransferItems));
+
+      this.setState({
+         isDragActive: allFilesAccepted,
+         borderStyle: 'dragBorder',
+         visibility: 'hide'
+         // isDragReject: !allFilesAccepted
+      });
+
+      if (allFilesAccepted) {
+         this.successBorder()
+      } else {
+         this.warningBorder();
+      }
+  }
+
+   /**
+      * Handle the drag over event
+      * @param  {Event]} event
+   */
+   onDragOver (e) {
+      e.preventDefault();
+      e.stopPropagation();
+      e.dataTransfer.dropEffect = 'copy';
+
+      return false;
+   }
+
+   /**
+      * Handle the drag leave event
+   */
+   onDragLeave (e) {
+      e.preventDefault();
+
+      // Only deactivate once the dropzone and all children was left.
+      if (--this.enterCounter > 0) {
+         return;
+      }
+
+      this.setState({
+         isDragActive: false,
+         isDragReject: false,
+         borderStyle: '',
+         visibility: 'show'
+      });
+
+      if (this.props.onDragLeave) {
+         this.props.onDragLeave.call(this, e);
+      }
+
+      if(this.state.imageFile){
+         // console.log('onDragLeave');
+         return;
+      }
+  }
+
+   /**
+      * Handle the drop event
+      * @param  {Event]} event
+   */
+   onDrop (e) {
+      e.preventDefault();
+      this.setState({
+         isDragActive: false,
+         borderStyle: '',
+         visibility: 'show'
+      });
+      this.handleChange.call(this, e);
+   }
+
+   /**
+      * Handle the click event on drop container
+   */
+   // onClick () {
+   //    console.log('on click called');
+   //    this.refs.fileInput.myTextInput.click();
+   // }
+
+   /**
+      * Handle the image state
+      * If image is available, will create a URL obj
+      * And modify the state value
+      * @param  {Event]} event
+   */
+   handleChange (event) {
+      let e = event.nativeEvent, imageFile;
+
+      const dataTransferItems = this.extractData(event);
+      const allFilesAccepted = this.fileAccepted(Array.prototype.slice.call(dataTransferItems));
+
+      // Check to make sure the dropped file
+      // is in image format - image/*
+      if(allFilesAccepted){
+         // Setting the state image source
+         // With drag & drop event
+         if(e.dataTransfer) {
+            let evt = e.dataTransfer.files[0];
+
+            if(evt !== undefined) {
```

Comment:
As a best practice, we try not to use the `undefined` variable as it could actually be defined. `typeof event !== 'undefined'` is a slightly more reliable way to check to see if event is defined.